src/contexts/StateContextProvider.tsx

```diff
@@ -246,14 +246,15 @@ const updateUrlIfEventOnPage = (
 // eslint-disable-next-line max-lines-per-function, max-statements
 const StateContextProvider = ({children}: StateContextProviderProps) => {
     const {postPopUp} = useContext(NotificationContext);
-    const {filePath, logEventNum} = useContext(UrlContext);
+    const {filePath, isCaseSensitive, isRegex, logEventNum, queryString} = useContext(UrlContext);
 
     // States
     const [exportProgress, setExportProgress] =
         useState<Nullable<number>>(STATE_DEFAULT.exportProgress);
+    const [fileName, setFileName] = useState<string>(STATE_DEFAULT.fileName);
+    const [isFileLoaded, setIsFileLoaded] = useState<boolean>(false);
```

Comment:
Instead of adding this state variable, can we simply run the query (if applicable) whenever `WORKER_RESP_CODE.LOG_FILE_INFO` is received?